client/shared/dev/customMochaSpecReporter.js

```diff
@@ -29,26 +30,35 @@ class SpecFileReporter extends mocha.reporters.Spec {
     }
 
     if ('BUILDKITE_LABEL' in process.env) {
-      this.title = process.env.BUILDKIATE_LABEL
+      this.title = process.env.BUILDKITE_LABEL || 'placeholder'
     }
 
     if (this.buildkite === true && typeof process.env.BUILDKITE_LABEL === undefined) {
-      console.warn(
-        `In Buildkite but BUILDKITE_LABEL not found in environment. Using title '${this.title || 'placeholder'}'`
-      )
+      console.info(`In Buildkite but BUILDKITE_LABEL not found in environment. Using title '${this.title}'`)
     }
   }
 
+  safeClose(stream) {
+    stream.close()
+    return new Promise((resolve, reject) => {
+      stream.once('close', () => {
+        resolve()
+      })
+    })
```

Comment:
It might be valuable to propagate errors from the `close` call to the error handler in the `epilogue` method.

```suggestion
    return new Promise((resolve, reject) => {
      stream.close(error => {
        if (error) {
          reject(error)
        }
        
        resolve()
      })
    })
```